editor/src/components/editor/store/dispatch.tsx

```diff
@@ -87,6 +102,149 @@ import {
   reduxDevtoolsSendActions,
   reduxDevtoolsUpdateState,
 } from '../../../core/shared/redux-devtools'
+import {
+  getSavedCodeFromTextFile,
+  getUnsavedCodeFromTextFile,
+} from '../../../core/model/project-file-utils'
+
+export interface WriteProjectFileChange {
+  type: 'WRITE_PROJECT_FILE'
+  fullPath: string
+  projectFile: ProjectFile
+}
+
+export function writeProjectFileChange(
+  fullPath: string,
+  projectFile: ProjectFile,
+): WriteProjectFileChange {
+  return {
+    type: 'WRITE_PROJECT_FILE',
+    fullPath: fullPath,
+    projectFile: projectFile,
+  }
+}
+
+export interface DeletePathChange {
+  type: 'DELETE_PATH'
+  fullPath: string
+  recursive: boolean
+}
+
+export function deletePathChange(fullPath: string, recursive: boolean): DeletePathChange {
+  return {
+    type: 'DELETE_PATH',
+    fullPath: fullPath,
+    recursive: recursive,
+  }
+}
+
+export interface EnsureDirectoryExistsChange {
+  type: 'ENSURE_DIRECTORY_EXISTS'
+  fullPath: string
+}
+
+export function ensureDirectoryExistsChange(fullPath: string): EnsureDirectoryExistsChange {
+  return {
+    type: 'ENSURE_DIRECTORY_EXISTS',
+    fullPath: fullPath,
+  }
+}
+
+export type ProjectChange = WriteProjectFileChange | DeletePathChange | EnsureDirectoryExistsChange
+
+export function collateProjectChanges(
```

Comment:
I think this 200-300 lines of new code here are enough to warrant a dedicated file!

Reply:
Not new code. :)

Reply:
_here_ it's new code :P 

but alas, my point was that I think this burdens dispatch.ts and instead could live in a dedicated file

Reply:
Sure, I'll try to get as much of this into another file.

Reply:
Fixed in a2eaa373.